### PR TITLE
Prepare for the merge /usr/sbin→/usr/bin merge

### PR DESCRIPTION
**Pull request by keszybz**
Also finish the merge of /bin→/usr/bin and /sbin/→/usr/sbin.

### CODE REVIEW

**Comment by keszybz** [comment]
This is fully untested.

**Comment by keszybz** [comment]
I found a few spots where there were duplicated rules after sbin patterns were replaced by bin.

**Comment by keszybz** [comment]
I'm not sure why the CI fails. I built the rpms locally and they seem fine.
As a test, I installed the updated policy packages on a F40 VM and it boots fine and selinux seems to work.
I'll test on a system with merged-sbin tomorrow.

**Comment by keszybz** [comment]
OK, this has been tested properly now. I found a few more spots where there were duplicate (identical) rules after the substitutions. This is all fixed now, except for dkim and qemu, which had duplicate rules in two different files already before. This doesn't seem to cause problems, so I left it as is.

I built a package for rawhide with those patches, and installed it in a fresh Cloud Rawhide image (Fedora-Cloud-Base-Generic.x86_64-Rawhide-20240415.n.0.qcow2). There are no AVCs after a reboot and everything seems to work fine. After that, I installed a bunch of packages with merged-sbin (from https://copr.fedorainfracloud.org/coprs/zbyszek/merged-sbin/), and rebooted, and it seems that the selinux policy works as expected. (There are some other problems, but related to missing /sbin paths, not any selinux problems).

So I think this works as expected and could be merged.

**Comment by keszybz** [comment]
https://src.fedoraproject.org/rpms/selinux-policy/pull-request/409 is the pull request against dist-git that I was testing.

**Comment by keszybz** [comment]
@zpytela I'd appreciate a review.

File: config/file_contexts.subs_dist [hunk 1]
```diff
@@ -19,4 +19,6 @@
 /usr/local/lib64 /usr/lib
 /usr/local/lib /usr/lib
 /var/run/lock /var/lock
+/bin /usr/bin
 /sbin /usr/sbin
```
Comment (zpytela) [inline config/file_contexts.subs_dist]: When /usr/sbin changes to be a subject of equivalency, this target needs to change, too. Even though these setting currently do not apply, we plan to move most of settings from dist-git to github.

File: policy/modules/admin/usermanage.fc [hunk 2]
```diff
@@ -1,35 +1,27 @@
-ifdef(`distro_gentoo',`
-/bin/passwd		--	gen_context(system_u:object_r:passwd_exec_t,s0)
-')
+/usr/lib/cracklib_dict.* --	gen_context(system_u:object_r:crack_db_t,s0)
```
Comment (zpytela) [inline policy/modules/admin/usermanage.fc]: I guess it was not intentional to move this line here, we try to follow alphabetical order whenever possible.

File: policy/modules/admin/su.fc [hunk 3]
```diff
@@ -1,6 +1,3 @@
-
-/bin/su			--	gen_context(system_u:object_r:su_exec_t,s0)
-
 /usr/(local/)?bin/ksu	--	gen_context(system_u:object_r:su_exec_t,s0)
```
Comment (zpytela) [inline policy/modules/admin/su.fc:1]: I guess this line also deserves a change.

File: policy/modules/contrib/milter.fc [hunk 4]
```diff
@@ -1,12 +1,14 @@
 /etc/mail/dkim-milter/keys(/.*)?        gen_context(system_u:object_r:dkim_milter_private_key_t,s0)
 
-/usr/sbin/dkim-filter           --      gen_context(system_u:object_r:dkim_milter_exec_t,s0)
-/usr/sbin/opendkim      --  gen_context(system_u:object_r:dkim_milter_exec_t,s0)
-/usr/sbin/opendmarc     --  gen_context(system_u:object_r:dkim_milter_exec_t,s0)
-/usr/sbin/milter-greylist	--	gen_context(system_u:object_r:greylist_milter_exec_t,s0)
-/usr/sbin/sqlgrey       --      gen_context(system_u:object_r:greylist_milter_exec_t,s0)
-/usr/sbin/milter-regex				--	gen_context(system_u:object_r:regex_milter_exec_t,s0)
-/usr/sbin/spamass-milter	--	gen_context(system_u:object_r:spamass_milter_exec_t,s0)
+# those are duplicate with dkim.fc
```
Comment (zpytela) [inline policy/modules/contrib/milter.fc:3]: so why not remove them, someone may merge these two policies later

File: policy/modules/system/authlogin.fc [hunk 5]
```diff
@@ -24,34 +22,20 @@ HOME_DIR/\.google_authenticator~		gen_context(system_u:object_r:auth_home_t,s0)
...
 
-/sbin/pam_console_apply	 --	gen_context(system_u:object_r:pam_console_exec_t,s0)
-/sbin/pam_timestamp_check --	gen_context(system_u:object_r:pam_timestamp_exec_t,s0)
-/sbin/tcb_convert	--	gen_context(system_u:object_r:updpwd_exec_t,s0)
-/sbin/tcb_unconvert	--	gen_context(system_u:object_r:updpwd_exec_t,s0)
-/sbin/unix_chkpwd	--	gen_context(system_u:object_r:chkpwd_exec_t,s0)
-/sbin/unix_update	--	gen_context(system_u:object_r:updpwd_exec_t,s0)
-/sbin/unix_verify	--	gen_context(system_u:object_r:chkpwd_exec_t,s0)
-ifdef(`distro_suse', `
-/sbin/unix2_chkpwd	--	gen_context(system_u:object_r:chkpwd_exec_t,s0)
```
Comment (zpytela) [inline policy/modules/system/authlogin.fc:35]: Removing this line (or three) was intentional?

**Review by zpytela — COMMENTED** [review]
LGTM, just a few minor glitches, but can even be merged as is. Thank you for your effort.

File: policy/modules/system/authlogin.fc (hunk 5 above)
Comment (keszybz) [reply policy/modules/system/authlogin.fc:35]: Hmm, I thought that the file is listed below, but it's `unix_chkpwd`, not `unix2_chkpwd`. I'll add it back.

File: config/file_contexts.subs_dist (hunk 1 above)
Comment (keszybz) [reply config/file_contexts.subs_dist]: Done.

File: policy/modules/contrib/milter.fc (hunk 4 above)
Comment (keszybz) [reply policy/modules/contrib/milter.fc:3]: I added a separate commit which does that.

File: policy/modules/admin/su.fc (hunk 3 above)
Comment (keszybz) [reply policy/modules/admin/su.fc:1]: I don't think so. It matches `/usr/local/bin/ksu` and `/usr/bin/ksu`, so no change needed?

Comment (zpytela) [reply policy/modules/admin/su.fc:1]: My bad, I thought /usr/local is also subject of equivalency, but it is not.

**Comment by zpytela** [comment]
So finally merging, thank you.

**Comment by keszybz** [comment]
Great, thanks!

**Pull request merged**